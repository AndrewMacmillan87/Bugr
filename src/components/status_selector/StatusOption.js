import React from "react";

import PropTypes from "prop-types";

const StatusOption = ({ value }) => {
    const classes = `${value} status-color`;

    return (
        <div className='status-option'>
            <div className={classes}>
                <p className='status-title'>{value}</p>
            </div>
        </div>
    );
};

StatusOption.propTypes = {
    value: PropTypes.string,
};

export default StatusOption;
