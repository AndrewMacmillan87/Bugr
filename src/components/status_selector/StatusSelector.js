import React from "react";

import StatusOption from "./StatusOption";

import PropTypes from "prop-types";

import "./status_selector.css";

export const StatusSelector = ({ status, onClick }) => {
    return status === "priority" ? (
        <div className='status-selector-container'>
            <i
                className='remove-selector fas fa-times-circle'
                onClick={() => onClick(false)}></i>
            <StatusOption value='Urgent' />
            <StatusOption value='High' />
            <StatusOption value='Medium' />
            <StatusOption value='Low' />
        </div>
    ) : (
        <div className='status-selector-container'>
            <i
                className='remove-selector fas fa-times-circle'
                onClick={() => onClick(false)}></i>
            <StatusOption value='Stuck' />
            <StatusOption value='Complete' />
            <StatusOption value='Progressing' />
            <StatusOption value='Todo' />
        </div>
    );
};

StatusSelector.propTypes = {
    status: PropTypes.string,
    onClick: PropTypes.func,
};

export default StatusSelector;
