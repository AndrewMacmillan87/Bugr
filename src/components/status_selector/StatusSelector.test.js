import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import StatusSelector from "./StatusSelector";
import StatusOption from "./StatusOption";

describe("StatusSelector component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<StatusSelector />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<StatusSelector />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("StatusSelector HTML elements", () => {
        it("should render an i element", () => {
            const wrapper = mount(<StatusSelector />);
            expect(wrapper.find("i").length).toEqual(1);
        });

        it("should render 4 StatusOption components", () => {
            const wrapper = mount(<StatusSelector />);
            expect(wrapper.find(StatusOption).length).toEqual(4);
        });
    });
});

describe("StatusOption component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<StatusOption />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<StatusOption />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("StatusOption HTML elements", () => {
        it("should render 2 div elements", () => {
            const wrapper = mount(<StatusOption />);
            expect(wrapper.find("div").length).toEqual(2);
        });

        it("should render a p element", () => {
            const wrapper = mount(<StatusOption />);
            expect(wrapper.find("p").length).toEqual(1);
        });
    });
});
