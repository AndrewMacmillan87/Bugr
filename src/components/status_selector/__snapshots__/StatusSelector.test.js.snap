// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusOption component Renders correctly 1`] = `
<div
  className="status-option"
>
  <div
    className="undefined status-color"
  >
    <p
      className="status-title"
    />
  </div>
</div>
`;

exports[`StatusSelector component Renders correctly 1`] = `
<div
  className="status-selector-container"
>
  <i
    className="remove-selector fas fa-times-circle"
    onClick={[Function]}
  />
  <div
    className="status-option"
  >
    <div
      className="Stuck status-color"
    >
      <p
        className="status-title"
      >
        Stuck
      </p>
    </div>
  </div>
  <div
    className="status-option"
  >
    <div
      className="Complete status-color"
    >
      <p
        className="status-title"
      >
        Complete
      </p>
    </div>
  </div>
  <div
    className="status-option"
  >
    <div
      className="Progressing status-color"
    >
      <p
        className="status-title"
      >
        Progressing
      </p>
    </div>
  </div>
  <div
    className="status-option"
  >
    <div
      className="Todo status-color"
    >
      <p
        className="status-title"
      >
        Todo
      </p>
    </div>
  </div>
</div>
`;
